### PR TITLE
# 161342849 paginate articles

### PR DESCRIPTION
**Pull request by wiztemple**
#### What does this PR do?
- Paginate articles

#### Description of tasks to be completed
- write a method to return all articles
- write a method to paginate articles
- test route to check functionalities

#### What are the Relevant Pivotal Tracker Stories?
- [#161342849](https://www.pivotaltracker.com/story/show/161342849)

#### How can this be manually be manually tested?
- git clone https://github.com/andela/haven-ah-backend.git.
- run npm install to install the npm packages.
- run npm test to run the unit and integration tests.
- To test on postman, run npm run start-dev
- On postman run GET http://localhost:5000/api/v1/articles

#### Screenshots(if appropriate)
screenshot 2018-11-01 at 6 51 43 pm
screenshot 2018-11-01 at 6 52 43 pm

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -3283,27 +3283,27 @@
       "dependencies": {
```
Comment (seyi-adeleke) [inline package-lock.json]: You didn't install a new package so this file shouldnt be changing

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

**Comment by PriscillaSam** [comment]
This looks great. Good job @wiztemple.

File: package-lock.json (hunk 1 above)
Comment (wiztemple) [reply package-lock.json]: I had a merge conflict, I had to rebase and run npm install.

File: server/utilities/httpResponse.js [hunk 2]
```diff
@@ -16,6 +17,9 @@ export const goodHttpResponse = (response, statusCode, message, data) => {
   if (data) {
     responseBody.data = data;
   }
+  if (meta) {
```
Comment (seyi-adeleke) [inline server/utilities/httpResponse.js]: Let's have another http helper paginatedHttpResponse. Lets keep this pure

File: server/controllers/article.js [hunk 3]
```diff
@@ -34,6 +34,27 @@ class Article {
...
+   * @param {object} response Response Object
+   * @returns {object} User Object
+   */
+  static async getArticles(request, response) {
+    try {
+      const limit = parseInt(request.query.limit, 10) || 10;
+      const page = parseInt(request.query.page, 10) || 1;
+
+      const articles = await articleRepo.getArticles(limit, page);
+      if (!articles) {
```
Comment (seyi-adeleke) [inline server/controllers/article.js:50]: a badhttpResponse should take either a 4xx/5xx as the status code.

File: server/tests/controllerTests/article.test.js [hunk 4]
```diff
@@ -31,3 +31,12 @@ describe('Post a new article:', () => {
       .equals('Article Created');
   });
 });
+
```
Comment (seyi-adeleke) [inline server/tests/controllerTests/article.test.js:34]: Can you test the endpoint with the actual query parameters please

File: server/tests/repositoryTests/article.test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import chai from 'chai';
+import articleRepo from '../../repository/articleRepository';
+// import data from '../utilities/mockData';
```
Comment (seyi-adeleke) [inline server/tests/repositoryTests/article.test.js]: remove commented out code

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

File: server/tests/repositoryTests/article.test.js (hunk 5 above)
Comment (wiztemple) [reply server/tests/repositoryTests/article.test.js]: okay. Thanks

File: server/utilities/httpResponse.js [hunk 6]
```diff
@@ -40,3 +43,12 @@ export const badHttpResponse = (response, statusCode, message, problem) => {
   }
   return response.status(statusCode).json(responseBody);
 };
+
```
Comment (seyi-adeleke) [inline server/utilities/httpResponse.js:44]: add jsdoc here

File: server/utilities/httpResponse.js (hunk 2 above)
Comment (seyi-adeleke) [reply server/utilities/httpResponse.js]: Please remove this if statement

File: server/utilities/httpResponse.js (hunk 6 above)
Comment (seyi-adeleke) [reply server/utilities/httpResponse.js:44]: ⏫

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**